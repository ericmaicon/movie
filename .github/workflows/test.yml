name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
