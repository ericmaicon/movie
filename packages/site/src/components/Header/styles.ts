import { Container } from 'react-grid-system';

import styled from 'styled-components';

export const Banner = styled.div`
  background: url('/images/header.jpg') no-repeat center center fixed;
  background-size: cover;
  min-height: 600px;
`;

export const NavBar = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding-top: 20px;
`;

export const TextContainer = styled(Container)`
  color: ${({ theme }) => theme.colors.white};
  margin-top: 150px;
  text-align: center;

  form {
    margin-top: 50px;
  }
`;

export const Button = styled.button`
  background: transparent;
  border: none;
  cursor: pointer;
  outline: none;
`;
