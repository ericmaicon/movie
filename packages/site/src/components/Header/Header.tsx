import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import { FaSearch } from 'react-icons/fa';

import { Button, Heading, Input, Text } from '@movie/uikit';
import { FormikValues, useFormik } from 'formik';

import * as Styled from './styles';

interface HeaderProps {
  // eslint-disable-next-line no-unused-vars
  onFilter: (values: FormikValues) => Promise<void>;
}

export default function Header({ onFilter, ...rest }: HeaderProps) {
  const formik = useFormik({
    initialValues: {
      search: '',
    },
    onSubmit: async (values) => {
      await onFilter(values);
    },
  });

  return (
    <Styled.Banner {...rest}>
      <Container>
        <Row>
          <Col sm={12}>
            <Styled.NavBar>
              <Button variant="secondary">Login</Button>
              <Button>Sign Up</Button>
            </Styled.NavBar>
          </Col>
        </Row>
      </Container>
      <Styled.TextContainer>
        <Row>
          <Col sm={12}>
            <Heading>Your favorite movies. Explained.</Heading>
            <Text>Figure out what happened. Then find out why.</Text>
          </Col>
        </Row>
        <Row>
          <Col sm={4} offset={{ sm: 4 }}>
            <form onSubmit={formik.handleSubmit} data-testid="form">
              <Input
                id="search"
                name="search"
                type="text"
                onChange={formik.handleChange}
                value={formik.values.search}
                startIcon={
                  <Styled.Button type="submit">
                    <FaSearch />
                  </Styled.Button>
                }
                placeholder="Search for a movie..."
              />
            </form>
          </Col>
        </Row>
      </Styled.TextContainer>
    </Styled.Banner>
  );
}
