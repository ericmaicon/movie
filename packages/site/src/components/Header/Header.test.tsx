import React from 'react';

import userEvent from '@testing-library/user-event';

import { fireEvent, render, waitFor } from '~/test/index';

import Header from './Header';

describe('Header', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <Header onFilter={jest.fn()} data-testid="header" />,
    );

    const headerElement = getByTestId('header');

    expect(headerElement).toBeInTheDocument();
    expect(headerElement).toMatchSnapshot();
  });

  it('should call onFilter after submit form', async () => {
    const onFilter = jest.fn();
    const { getByRole, getByTestId } = render(<Header onFilter={onFilter} />);

    const searchInput = getByRole('textbox');
    const formElement = getByTestId('form');

    userEvent.type(searchInput, 'Search');
    fireEvent.submit(formElement);
    await waitFor(() =>
      expect(onFilter).toHaveBeenCalledWith({
        search: 'Search',
      }),
    );
  });
});
