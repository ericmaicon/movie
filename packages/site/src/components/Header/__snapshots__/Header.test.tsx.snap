// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
.c2 {
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  border: none;
  color: #E9E8E7;
}

.c2:hover {
  color: #1C1C1C;
}

.c2:active {
  color: #1C1C1C;
}

.c2:focus {
  color: #1C1C1C;
}

.c2:disabled {
  color: #1C1C1C;
}

.c3 {
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  border: 1px solid #A62218;
  color: #A62218;
}

.c3:hover {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c3:active {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c3:focus {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c3:disabled {
  border: 1px solid #A8B4BE;
  color: #1C1C1C;
}

.c5 {
  margin: 0px;
  font-size: 2.5rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  color: #1C1C1C;
  padding-left: 10px;
}

.c10 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c10:disabled {
  cursor: not-allowed;
}

.c10::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c10::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c10:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c10::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E9E8E7;
  border: 1px solid #1C1C1C;
  box-sizing: border-box;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: horizontal;
  -ms-flex-direction: horizontal;
  flex-direction: horizontal;
  height: 30px;
}

.c6 {
  font-size: 0.2rem;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c0 {
  background: url('/images/header.jpg') no-repeat center center fixed;
  background-size: cover;
  min-height: 600px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-top: 20px;
}

.c4 {
  color: #E9E8E7;
  margin-top: 150px;
  text-align: center;
}

.c4 form {
  margin-top: 50px;
}

.c9 {
  background: transparent;
  border: none;
  cursor: pointer;
  outline: none;
}

@media (min-width:768px) and (max-width:1024px) {
  .c2 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c2 {
    font-size: 1rem;
  }
}

@media (min-width:768px) and (max-width:1024px) {
  .c3 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c3 {
    font-size: 1rem;
  }
}

@media (min-width:768px) and (max-width:1024px) {
  .c5 {
    font-size: 3rem;
  }
}

@media (min-width:1025px) {
  .c5 {
    font-size: 4rem;
  }
}

@media (min-width:768px) and (max-width:1024px) {
  .c10 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c10 {
    font-size: 1rem;
  }
}

@media (min-width:768px) and (max-width:1024px) {
  .c6 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c6 {
    font-size: 1rem;
  }
}

<div
  class="c0"
  data-testid="header"
>
  <div
    style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px; max-width: 960px;"
  >
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 100%; flex-grow: 0; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <nav
          class="c1"
        >
          <button
            class="c2"
          >
            Login
          </button>
          <button
            class="c3"
          >
            Sign Up
          </button>
        </nav>
      </div>
    </div>
    <span
      style="display: table; clear: both;"
    />
  </div>
  <div
    class="c4"
    style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px; max-width: 960px;"
  >
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 100%; flex-grow: 0; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <h1
          class="c5"
          size="1"
        >
          Your favorite movies. Explained.
        </h1>
        <p
          class="c6"
        >
          Figure out what happened. Then find out why.
        </p>
      </div>
    </div>
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 33.333333333333336%; flex-basis: 33.333333333333336%; flex-grow: 0; flex-shrink: 0; max-width: 33.333333333333336%; margin-left: 33.333333333333336%;"
      >
        <form
          data-testid="form"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <button
                class="c9"
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </button>
            </div>
            <input
              class="c10"
              id="search"
              name="search"
              placeholder="Search for a movie..."
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </div>
    <span
      style="display: table; clear: both;"
    />
  </div>
</div>
`;
